index.js

```diff
@@ -14,18 +14,16 @@
  \*───────────────────────────────────────────────────────────────────────────*/
 'use strict';
 
-var async = require('async'),
-  Setup = require('./setup'),
+var Promiz = require('./lib/promise'),
+  Configure = require('./lib/configure'),
+  Setup = require('./lib/setup'),
```

Comment:
my initial thought was we could just go ahead and expose the `Setup` variable as well (and leave the Nemo() constructor as is), but after more thought- it doesn't really matter- I kind of like it the way you have it.
The only reason I mentioned it is b/c it might simplify things if the function only accepts one signature (i.e. if someone wants to use Configure() and Setup() by themselves, they can- and Nemo() is just a nice wrapper around calling those 2 things).

It might be too late to enforce a single signature (since we already allow callbacks vs promises, etc).  Just wanted to see if we could get away from the complex argument processing.